pmd-core/src/main/java/net/sourceforge/pmd/Report.java

```diff
@@ -474,39 +477,89 @@ public boolean treeIsEmpty() {
      * Returns an iteration over the reported violations.
      *
      * @return an iterator
+     *
+     * @deprecated The {@link ReportTree} is deprecated
      */
+    @Deprecated
     public Iterator<RuleViolation> treeIterator() {
         return violationTree.iterator();
     }
 
+    /**
+     * @deprecated Use {@link #getViolations()}
+     */
+    @Deprecated
     @Override
     public Iterator<RuleViolation> iterator() {
         return violations.iterator();
     }
 
+
+    /**
+     * Returns an unmodifiable list of violations that were suppressed.
+     */
+    public List<SuppressedViolation> getSuppressedRuleViolations() {
+        return Collections.unmodifiableList(suppressedRuleViolations);
+    }
+
+    /**
+     * Returns an unmodifiable list of violations that have been
+     * recorded until now. None of those violations were suppressed.
+     */
+    public List<RuleViolation> getViolations() {
```

Comment:
We seem to keep the violations sorted by `RuleViolationComparator.INSTANCE`. Maybe that's worth mentioning, that the list of violations is actually a sorted list?